tests/src/FunctionalJavascript/CivicrmEntityViewsTestBase.php

```diff
@@ -0,0 +1,156 @@
+<?php declare(strict_types=1);
+
+namespace Drupal\Tests\civicrm_entity\FunctionalJavascript;
+
+use Behat\Mink\Exception\ElementNotFoundException;
+use Drupal\civicrm_entity\SupportedEntities;
+use Drupal\Core\Database\Database;
+use Drupal\Core\Url;
+
+abstract class CivicrmEntityViewsTestBase extends CivicrmEntityTestBase {
+
+  protected static $modules = [
+    'views',
+    'views_ui',
+  ];
+
+  protected static $civicrmEntityTypeId = NULL;
+
+  protected static $civicrmEntityPermissions = [];
+
+  protected function doInstall() {
+    parent::doInstall();
+
+    // The database information was added inside of our test environment,
+    // but it wasn't added to the Drupal settings to make it available for
+    // directly queries via Views.
+    // @todo This needs to be documented for all users wanting Views integration.
+    // @todo There is a workaround in CivicrmSql::init, that is why this is
+    //   commented out. This way we can test the workaround.
+    // @see \Drupal\Core\Installer\Form\SiteSettingsForm::submitForm
+    // @see \Drupal\Tests\civicrm\FunctionalJavascript\CiviCrmTestBase::changeDatabasePrefix
+    // @see \Drupal\civicrm_entity\Plugin\views\query\CivicrmSql::init
+    /*
+    $connection = Database::getConnection('default', 'civicrm_test')->getConnectionOptions();
+    $settings['databases']['civicrm_test']['default'] = (object) [
+      'value'    => [
+        'driver' => $connection['driver'],
+        'username' => $connection['username'],
+        'password' => $connection['password'],
+        'host' => $connection['host'],
+        'database' => $connection['database'],
+        'namespace' => $connection['namespace'],
+        'port' => $connection['port'],
+        // CiviCRM does not use prefixes.
+        'prefix' => '',
+      ],
+      'required' => TRUE,
+    ];
+    $this->writeSettings($settings);
+    */
+  }
```

Comment:
This was the original fix. But it prevents testing the backward compatibility layer. I wasn't sure how we should document this. I also wonder if we should test this via a Kernel test or something. Where this is defined and respected.